### ISSUE TITLE
Fix false positives when importing from module with `css` in name

### ISSUE THREAD

**Issue by OliverJAsh** [stylelint/stylelint#4706]
> Clearly describe the bug

There are false positives in JS/TS files when using a function imported from a module with `css` in the name.

> Which rule, if any, is the bug related to?

Several:

`stylelint.config.js`:
```javascript
module.exports = {
    rules: {
        'selector-max-type': 0,
        'selector-type-no-unknown': true,
        'block-no-empty': true,
        'block-closing-brace-newline-after': 'always',
        'property-no-unknown': true,
        'declaration-block-trailing-semicolon': 'always',
    },
};
```

> What code is needed to reproduce the bug?

`src/main.js`:
```javascript
import { fn2 } from './css';

const fn1 = (x) => x;
// no errors here
fn1({ styles: {}, key: 'height' });

// lots of errors here
fn2({ styles: {}, key: 'height' });
```

If I change the import to something which doesn't have `css` in the name, the errors go away.

> What stylelint configuration is needed to reproduce the bug?

Pasted above

> Which version of stylelint are you using?

13.3.2

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint ./src/main-js.js`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Not sure

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

```sh-session
$ stylelint ./src/main-js.js

src/main-js.js
 8:6   ✖  Expected "styles" to have no more than 0 type selectors   selector-max-type
 8:6   ✖  Unexpected unknown type selector "styles"                 selector-type-no-unknown
 8:14  ✖  Unexpected empty block                                    block-no-empty
 8:16  ✖  Expected newline after "}"                                block-closing-brace-newline-after
 8:18  ✖  Unexpected unknown property "key"                         property-no-unknown
 8:30  ✖  Expected a trailing semicolon                             declaration-block-trailing-semicolon
```

**Comment by jeddy3** [comment]
@OliverJAsh Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwATFwDMA7AJlwF8jMIVcByAOgHowBnN+gbgB1i-IxNnhIBGXAF5cACgAeASkkA+XLN7FmzXMQi4YmGpja4AFvph8x0gsICeAGxht4+CgBpcAaxi2X9M0gA5ibY9JTy6nyauPYQ2MYQhHoGWMZmmBbEJKTWuHaOzq4e3r4MAcGh4VwgbiBstsTYAIayCOAcALRIxB0AVmw14BBZQW3AfLi4PCCYAK4F0y7jxJOT02wwjmDYWB0oLR3YtmgWCLgADG4Tq+ubMNu7RycdOh2zxJ46AO78Z9hzMCuKxuIAARrEwJ4XhAOjB0EdFrh-rNAdc1mCIVCwLE2N1Ah1QZgmmAYC8YF97N1SU1CNh9Ijpk17F8mrY2NMgaspiA0DQTjhbNC3h9vr8XMjUcD0bBsU0idgkMMCZjDkSkJTiPiNigkJBYmLuUyWWzptcKHwKCAKEA).

The issue is likely with [the parser](https://github.com/stylelint/postcss-css-in-js) stylelint uses [to support CSS-in-JS](https://stylelint.io/about/syntaxes). I've added this issue to https://github.com/stylelint/stylelint/issues/4574, which is a growing list of problems with the syntax that we need help to fix.

Please consider contributing to the parser if you have time.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
